README.rst

```diff
@@ -0,0 +1,212 @@
+
+* There are no access controls for the API.
+* All email originates from a single email account.
+* Email is sent in plain text only.
+* Email is limited to a single recipient.
+* It is possible that a message may be lost if something horrible happens to
+  the rabbit queue, although we should be able to query the database for PENDING
+  messages and requeue them in case of rabbit disaster.
+
+API
+---
+
+POST /messages
+~~~~~~~~~~~~~~
+
+Send an email message.
+
++-----------+---------------------------------+
+| Parameter | Description                     |
++===========+=================================+
+| to        | Email address of the recipient. |
++-----------+---------------------------------+
+| subject   | Message subject.                |
++-----------+---------------------------------+
+| text      | Body of the message.            |
++-----------+---------------------------------+
+
+Request
++++++++
+
+.. code-block:: javascript
+
+    POST /messages
+    Headers:
+        Content-Type: application/json
+
+    Content:
+    {
+      "to": "someone@example.com",
+      "subject": "Example subject",
+      "text": "Example text."
+    }
+
+Response
+++++++++
+
+.. code-block:: javascript
+
+   201 CREATED
+   Headers:
+     Location: http://localhost:5000/messages/d39aca80-bf8f-42db-9fec-1828cfaf01fd
+
+   Content:
+   {
+     "id": "d39aca80-bf8f-42db-9fec-1828cfaf01fd",
+     "status": "PENDING",
+     "subject": "Example subject"
+   }
+
+GET /messages
+~~~~~~~~~~~~~
+
+Get a list of all messages odered from newest to oldest.  The list
+can be paginated by using the `limit` and `offset` parameters.
+
++-----------+-----------------------------------------------------+
+| Parameter | Description                                         |
++===========+=====================================================+
+| limit     | Maximum number of messages to return (default: 10). |
++-----------+-----------------------------------------------------+
+| offset    | Starting index for the list (default: 0).           |
++-----------+-----------------------------------------------------+
+
+Request
++++++++
+
+.. code-block:: javascript
+
+   GET /messages?offset=0&limit=10
+   Headers:
+      Accept: application/json
+
+Response
+++++++++
+
+.. code-block:: javascript
+
+   200 OK
+
+   Content:
+   {
+     "messages": [
+       {
+         "id": "4bef3ffd-d0ad-4037-abf3-062e9ceff507",
+         "status": "DELIVERED",
+         "subject": "Example subject"
+       }
+     ],
+     "count": 1
+   }
+
+GET /messages/{message_id}
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Get a single message status.
+
+Request
++++++++
+
+.. code-block:: javascript
+
+   GET /messages/4bef3ffd-d0ad-4037-abf3-062e9ceff507
+
+Response
+++++++++
+
+.. code-block:: javascript
+
+   200 OK
+
+   Content:
+   {
+     "id": "4bef3ffd-d0ad-4037-abf3-062e9ceff507",
+     "status": "DELIVERED",
+     "subject": "Example subject"
+   }
```

Comment:
Thanks!  Since this is a tiny API, I manually wrote the docs.  For a large project I would prefer to use [RAML](http://raml.org/) or [Swagger](http://swagger.io/)